From 866d81882e84424dd7dacf01731115ea945a5d34 Mon Sep 17 00:00:00 2001
From: Tianmi Chen <tianmi.chen@intel.com>
Date: Tue, 30 Apr 2013 14:06:14 +0800
Subject: Some mkv clips can't be recognized in Gallery

BZ: 48944

The Matroska Extractor fails to recognize the
"V_MS/VFW/FOURCC" Codec ID. The patch adds support
for this CodecID.

Original patch: R3 30130

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Change-Id: I6524c84a2ff90ad96a3d50159995e9fff83f97d0
Orig-Change-Id: I044f9de803cb9b1b00c064186e30bca1eb3c8496
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: wfeng6 <wei.feng@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index 4640c1c..4869c10 100644
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -873,6 +873,8 @@ void MatroskaExtractor::addTracks() {
                     meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_VP8);
                 } else if (!strcmp("V_VP9", codecID)) {
                     meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_VP9);
+                } else if (!strcmp("V_MS/VFW/FOURCC", codecID)) {
+                    meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_H263);
                 } else {
                     ALOGW("%s is not supported.", codecID);
                     continue;
-- 
1.7.4.1

